contracts/transfer/src/tree.rs

```diff
@@ -57,8 +55,11 @@ impl Tree {
         notes: I,
     ) {
         for note in notes {
-            let leaf = TreeLeaf { block_height, note };
-            self.push(leaf);
+            // skip transparent notes with a value of 0
+            if !note.value(None).is_ok_and(|value| value == 0) {
+                let leaf = TreeLeaf { block_height, note };
```

Comment:
nitpick: Could avoid creating a variable here like
```rust
self.push(TreeLeaf {
  block_height,
  note,
});
```

Reply:
But the current version is more compact 
2 lines of code against 4 suggested